server/routes/user.js

```diff
@@ -4,66 +4,75 @@ const User = require('../database/models/user');
 const passport = require('../passport');
 
 router.post('/', (req, res) => {
-  console.log('user signup');
+    console.log('user signup');
 
-  const { username, password, name, email } = req.body;
-  // ADD VALIDATION
-  User.findOne({ username: username }, (err, user) => {
-    if (err) {
-      console.log('User.js post error: ', err);
-    } else if (user) {
-      res.json({
-        error: `Sorry, already a user with the username: ${username}`,
-      });
-    } else {
-      const newUser = new User({
-        username: username,
-        password: password,
-        name: name,
-        email: email,
-      });
-      newUser.save((err, savedUser) => {
-        if (err) return res.json(err);
-        res.json(savedUser);
-      });
-    }
-  });
+    const { username, password, name, email } = req.body;
+    // ADD VALIDATION
+    User.findOne({ username: username }, (err, user) => {
+        if (err) {
+            console.log('User.js post error: ', err);
+        } else if (user) {
+            res.json({
+                error: `Sorry, already a user with the username: ${username}`,
+            });
+        } else {
+            const newUser = new User({
+                username: username,
+                password: password,
+                name: name,
+                email: email,
+            });
+            newUser.save((err, savedUser) => {
+                if (err) return res.json(err);
+                res.json(savedUser);
+            });
+        }
+    });
+});
+
+router.get('/data/:username', (req, res, next) => {
+    console.log("this is for user: " + req.params.username)
+    User.findOne({ username: req.params.username })
+        .then(function (userData) {
+            console.log(userData);
+            res.json(userData);
+        });
```

Comment:
[discussion] Just like in React when making API calls, we should also build error handling into our DB queries on the server. In this case, if we don't find the user, we should return `404 Not found` back to the client app.
```js
User.findOne({ username: req.params.username }).then(
  (userData) => res.json(userData),
  (error) => res.sendStatus(404)
);
```

Reply:
This makes sense (and thanks for writing the code 👍 ). 

Related:  I need to figure out how to make the route inaccessible if users are not logged in. Currently, you can run the route, and the API call reads the username from local storage to pull the data from the server. once AWS Cognito is set up, I'll need to rewrite this route to get the username (or ID or something else) from the token so that info can be passed to as a parameter.

Reply:
For now, don't worry about logged-in verification. We'll address this very problem once Cognito is implemented. How? Each API request for user data will need to be accompanied with an `Authorization` header with an access token. We validate the token is valid and matches the username requested before returning any data.